cli/tests/test_next_prev_commands.rs

```diff
@@ -70,30 +71,76 @@ fn test_prev_simple() {
     test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "first"]);
     test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "second"]);
     test_env.jj_cmd_ok(&repo_path, &["commit", "-m", "third"]);
+    // @ is an empty discardable commit created by `jj commit`.
+    insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
+    @  zsuskulnrvyr
+    ◉  kkmpptxzrspx third
+    ◉  rlvkpnrzqnoo second
+    ◉  qpvuntsmwlqt first
+    ◉  zzzzzzzzzzzz
+    "###);
+
     let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["prev"]);
-    // The working copy commit is now a child of "second".
+    // The working copy commit is now a child of "second", even though the
+    // commit before our empty working copy commit is actually "third". This is
+    // because we skip the empty discardable @ commit when deciding how far to
+    // move backwards.
     insta::assert_snapshot!(stdout, @"");
     insta::assert_snapshot!(stderr, @r###"
-    Working copy now at: mzvwutvl f7a0f876 (empty) (no description set)
+    Working copy now at: royxmykx 5647d685 (empty) (no description set)
     Parent commit      : rlvkpnrz 5c52832c (empty) second
     "###);
 }
 
+#[test]
+fn test_prev_non_discardable_working_copy() {
+    // Move from third => second.
```

Comment:
Maybe you mean "second => first" ?

"third" is the editing working-copy commit, which isn't considered the current position.

Reply:
Are you saying that `@` isn't considered the "current position" for `jj prev`, or that `jj` in general defines the "current position" as something other than `@`? I actually might not know the definition. What is it?

I understand now that `jj prev` moves relative to `@-`, so I updated the comment accordingly.

Reply:
> Are you saying that `@` isn't considered the "current position" for `jj prev`, or that `jj` in general

In `jj prev` and `jj next` without `--edit`. New comment looks better, thanks.